model/histogram/float_histogram.go

```diff
@@ -504,6 +599,10 @@ func (h *FloatHistogram) DetectReset(previous *FloatHistogram) bool {
 	if h.Count < previous.Count {
 		return true
 	}
+	if h.HasCustomBounds() != previous.HasCustomBounds() {
+		// TODO(zenador): correct?
```

Comment:
This is tricky, this means going from custom buckets to exponential buckets or the other way. Also the overall Count has not decreased.

In practice this would happen because the application was restarted with native histograms now included , in which case hopefully the Count (or even better created timestamp) reflects it.
(Since you cannot have custom histograms without native histograms enabled at scrape and we prefer exponential histograms, it cannot happen that we suddenly scrape and send native histograms.)

If the Count does not reflect the change, we'd have to compare the buckets themselves. But we don't have that capability in this code, so we just don't know. Also any bucket compare will include some interpolation to align the buckets so we may never know for sure.

I would just trust `Count` at this point and return `false` .

Reply:
On second thought you can only get different schema if you restart the application so `true` is fine

Reply:
Discussed in histograms sync meeting. Agreed on returning `true`. In general it should matter actually because:

When we write chunks a change in schema starts a new chunk and we should set "UNKOWN" reset hint for the new chunk.

When evaluating PromQL expressions, it should be treated currently as is we went from float to native histograms, meaning no result due to incompatibility.